sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -37,7 +46,115 @@ def test_transform_linear_model():
 
 def test_invalid_input():
     clf = SGDClassifier(alpha=0.1, n_iter=10, shuffle=True, random_state=None)
+    for threshold in ["gobbledigook", ".5 * gobbledigook"]:
+        model = SelectFromModel(clf, threshold=threshold)
+        model.fit(iris.data, iris.target)
+        assert_raises(ValueError, model.transform, iris.data)
+
+
+def test_validate_estimator():
```

Comment:
I did not add this, but I think the point is that the input estimator is unchanged?